docs/software/apptainer.md

```diff
@@ -0,0 +1,233 @@
+     
+     module load openmind8/apptainer/1.1.7      # load an apptainer module
+     singularity exec my-image.sif python my-code.py   # Run the program 
+     ```
+
+The last line is a command to run a Python program uisng Singularity.  
+
+Submit the job script using `sbatch`,
+```
+sbatch job.sh
+```
+
+### More on using Singularity
+
+In many cases, GPUs are needed to accelerate programs. As the GPU driver is installed on the host, use the flag `--nv` to pass necessary GPU driver libraries into the container, so that the program can "see" the GPUs in the container. 
+
+Check if GPUs are available in a container,
+```
+singularity exec --nv my-image.sif nvidia-smi
+```
+
+Here is an exmaple to run Python programs on GPUs.
+```
+singularity exec --nv my-image.sif python my-code.py  
+```
+
+By default, the home directory and the `/tmp` directory are bound to the container. If your programs read/write data files in other directories (e.g. `/path/to/data`), bind them to the container using the flag `-B`,
+```
+singularity exec -B /path/to/data my-image.sif python my-code.py  
+```
+
+In summary, a commonly used syntax to run a program with Singularity is the following,
+```
+singularity exec [--nv] [-B <path-to-data>] <image-name> <executable-name> [source-code-name]
+```
+The terms in `<>` are must-needed while the term in `[]` is optional, dependeing on use cases. 
+
+=== "Engaging"
+
+     Here is an example job script to run a python program with a GPU and data files saved in `/nobakcup1` or `/pool001` directories,
+     ```
+     #!/bin/bash                      
+     #SBATCH -t 01:30:00         # walltime = 1 hours and 30 minutes
+     #SBATCH -N 1                # one node
+     #SBATCH -n 2                # two CPU cores
+     #SBATCH --gres=gpu:1        # one GPU
+     #SBATCH -p sched_mit_psfc_gpu_r8     # a partition with Rocky 8 nodes
+
+     module load apptainer/1.1.7-x86_64 squashfuse/0.1.104-x86_64   # load modules
+     singularity exec --nv -B /nobakcup1,/pool001 my-image.sif python my-code.py   # Run the program
+     ```
+
+=== "OpenMind"
+
+     Here is an example job script to run a Python program with a GPU and data files saved in `/om` or `/om2` directories,
+     ```
+     #!/bin/bash                      
+     #SBATCH -t 01:30:00             # walltime = 1 hours and 30 minutes
+     #SBATCH -N 1                    # one node
+     #SBATCH -n 2                    # two CPU cores
+     #SBATCH --gres=gpu:1            # one GPU
+     #SBATCH --constraint=rocky8     # nodes with Rocky 8 OS
+
+     module load openmind8/apptainer/1.1.7        # load an apptainer module
+     singularity exec --nv -B /om,om2 my-image.sif python my-code.py  # Run the program
+     ```
+
+## Build Singularity images
+
+In the previous section, it is assumed that all needed packages have been installed in the image. If some needed packages do not exist in the image, users need to build a new image. 
```

Comment:
We may want to add a note here that says building containers is currently not supported on Engaging.